docker-image-scanner.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import sys
+import re
+import subprocess
+
+from typing import List, Dict
+
+
+def load_regex_patterns(regex_file: str) -> List[str]:
+    """Load regex patterns from a file."""
+    try:
+        with open(regex_file, "r") as f:
+            return [line.strip() for line in f if line.strip()]
+    except FileNotFoundError:
+        print(f"Error: Regex file '{regex_file}' not found", file=sys.stderr)
+        sys.exit(1)
+
+
+def format_image_path(image_path: str) -> str:
+    """Format the image path to use file:/// prefix with absolute path."""
+    abs_path = os.path.abspath(image_path)
+    return f"file://{abs_path}"
+
+
+def scan_docker_image(image_path: str) -> List[Dict]:
+    """Scan Docker image using trufflehog and return JSON results."""
+    try:
+        formatted_path = format_image_path(image_path)
+        cmd = ["trufflehog", "docker", "--image", formatted_path, "--json"]
```

Comment:
Does this do `--no-verification` by default or do we need to specify it so that even unverifiable secrets are also caught?

Reply:
we need to pass `--no-verification` explicitly. adding it.